repose-aggregator/components/services/open-tracing-service/open-tracing-service-api/src/main/resources/META-INF/schema/config/opentracing.xsd

```diff
@@ -1,315 +0,0 @@
-                    <html:p>
-                        Service name to use.  Will identify this service (e.g. identity-repose)
-                    </html:p>
-                </xs:documentation>
-            </xs:annotation>
-        </xs:attribute>
-
-        <xs:attribute name="flush-interval-ms" type="PositiveInt" default="1000">
-            <xs:annotation>
-                <xs:documentation>
-                    <html:p>
-                        Flush interval in milliseconds.  How often to send span data to tracer.  Defaults to 1 second.
-                    </html:p>
-                </xs:documentation>
-            </xs:annotation>
-        </xs:attribute>
-
-        <xs:attribute name="max-buffer-size" type="PositiveInt" default="10000">
-            <xs:annotation>
-                <xs:documentation>
-                    <html:p>
-                        Maximum number of spans in buffer.  Defaults to 10,000.
-                    </html:p>
-                </xs:documentation>
-            </xs:annotation>
-        </xs:attribute>
-
-        <xs:attribute name="log-spans" type="xs:boolean" default="true">
-            <xs:annotation>
-                <xs:documentation>
-                    <html:p>
-                        Optionally sets up logger for all spans.  Defaults to true (helps with debugging).
-                    </html:p>
-                </xs:documentation>
-            </xs:annotation>
-        </xs:attribute>
-
-        <xs:attribute name="sender-protocol" type="jaegerSenderProtocol" default="http">
-            <xs:annotation>
-                <xs:documentation>
-                    <html:p>
-                        Sender protocol:
-
-                        - udp (uses com.uber.jaeger.senders.UdpSender class to send all traffic over udp)
-                        - http (uses com.uber.jaeger.senders.HttpSender class to send all traffic over http)
-
-                        Default is http
-                    </html:p>
-                </xs:documentation>
-            </xs:annotation>
-        </xs:attribute>
-
-        <xs:attribute name="agent-host" type="xs:string" default="localhost">
-            <xs:annotation>
-                <xs:documentation>
-                    <html:p>
-                        Tracer agent host.  Defaults to localhost.  Used with udp protocol.  Ignored otherwise.
-                    </html:p>
-                </xs:documentation>
-            </xs:annotation>
-        </xs:attribute>
-
-        <xs:attribute name="agent-port" type="PositiveInt" default="6831">
-            <xs:annotation>
-                <xs:documentation>
-                    <html:p>
-                        Tracer agent port.  Defaults to 6831.  Used with udp protocol.  Ignored otherwise.
-                    </html:p>
-                </xs:documentation>
-            </xs:annotation>
-        </xs:attribute>
-
-
-        <xs:attribute name="collector-endpoint" type="xs:string" default="http://localhost:14268">
```

Comment:
It seems like we lost this after the update? It looks like the agent host/port was conflated with the collector host/port.

Reply:
Is that the real intent or an oversight?

Reply:
Collector =! agent .  Agent would be a thing living alongside Repose that you can push to via UDP while Collector is something remote you can push to via TCP.

Out of the box, we should support both (right now our implementation of Jaeger does not support agents but that should change soon ™️  as using agents is best practice)

Reply:
A great question! At first I thought it was an oversight, but now I'm thinking it was intentional. My initial thinking was that you would always have an agent _and_ collector, and you would only even "talk" to the agent, but it appears that you only ever talk to one depending on the transport protocol being used. So it's an either/or thing. Either you configure a host/port for an agent if you're using UDP (thrift) as the transport, or you configure a host/port for the collector if you're using HTTP as the transport.
See: https://jaeger.readthedocs.io/en/latest/architecture/#agent

In either case, you only have to specify one endpoint to push data to, so I suspect that is why Adrian collapsed them. Plus, this is all Jaeger specific, so it was moved into the Jaeger configuration as well.